### PR TITLE
Fix withdraw() methods in nullifier contracts for not validating the root

### PR DESCRIPTION
**Pull request by jimthematrix**

### CODE REVIEW

File: solidity/contracts/zeto_anon_nullifier.sol [hunk 1]
```diff
@@ -122,6 +122,7 @@ contract Zeto_AnonNullifier is
         uint256 root,
         Commonlib.Proof calldata proof
     ) public {
+        validateTransactionProposal(nullifiers, [output, 0], root);
```
Comment (Chengxuan) [inline solidity/contracts/zeto_anon_nullifier.sol:125]: Should we also add a test for this scenario?

**Review by Chengxuan — COMMENTED** [review]
@jimthematrix 👍 good catch, added a comment about testing.

Also, a minor point about the three functions in withdraw seem can be grouped into a _withdraw() function into zeto_nullifier like _mint.

**Review by EnriqueL8 — COMMENTED** [review]
This looks good to me - agree on the point above from Cheng but understand they are similar flows and tests have been included

**Review by Chengxuan — APPROVED** [review]
Thanks @jimthematrix 

It seems the security hole had to be patched to multiple contracts, so considering refactoring the check into the shared library will make it more reusable. But it's a minor comment for you to consider.

**Comment by jimthematrix** [comment]
Regarding the idea to push the calls in the `withdraw()` method to a library, the issue is that the 3 function calls already came from 2 parent classes:

```
        validateTransactionProposal(inputs, [output, 0], proof);    <<-- this is from Zeto_Base/Zeto_Nullifier
        _withdraw(amount, inputs, output, proof);                          <<-- this is from Zeto_Fungible
        processInputsAndOutputs(inputs, [output, 0]);                  <<-- this is from Zeto_Base/Zeto_Nullifier
```

**Pull request merged**